File: Python3/Sort Colors.py
#1
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        m = {}
        for n in nums:
            if n in m:
                m[n] = m[n] + 1
            else:
                m[n] = 1
        index = 0
        for i in range(3):
            for j in range(m.get(i, 0)):
                nums[index] = i
                index = index + 1

#2
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        red, white, blue = 0, 0, len(nums)-1
    
        while white <= blue:
            if nums[white] == 0:
                nums[red], nums[white] = nums[white], nums[red]
                white += 1
                red += 1
            elif nums[white] == 1:
                white += 1
            else:
                nums[white], nums[blue] = nums[blue], nums[white]
                blue -= 1

File: Golang/Sort Colors.go
func sortColors(nums []int)  {
    nZeros, nOnes, nTwos := 0, 0, 0
    
    for i := 0; i < len(nums); i++ {
        if nums[i] == 0 {
            nZeros++
        } else if nums[i] == 1{
            nOnes++
        } else {
            nTwos++
        }
    }
    
    i := 0

    for nZeros > 0 {
        nums[i] = 0
        nZeros--
        i++
    }
    
    for nOnes > 0 {
        nums[i] = 1
        nOnes--
        i++
    }
    
    for nTwos > 0 {
        nums[i] = 2
        nTwos--
        i++
    }
}